### PR TITLE
[expo-firebase-analytics] Updates Regex To Allow Numeric Chars

### PR DESCRIPTION
**Pull request by thorbenprimke**
# Why

The `eventName` and `userPropertyName` for a firebase event support alphanumeric values but the regex was not allowing numeric characters. 

# How

Updates regex and adds additional check to ensure the first char is alphabetic. 

# Test Plan

Added unit tests with expected results. Saw them fail. Updated logic and now they pass.

### CODE REVIEW

**Comment by IjzerenHein** [comment]
Thanks for this PR Thorben!

This seems like a duplicate of #8391, but this PR seems more complete. Although I think the regexp solution by @frantic is a little bit cleaner. Really appreciate the unit-tests here 🙏

File: packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts [hunk 1]
```diff
@@ -189,7 +189,8 @@ class FirebaseAnalyticsJS {
       !eventName.length ||
       eventName.length > 40 ||
       eventName[0] === '_' ||
-      !eventName.match(/^[A-Za-z_]+$/) ||
+      eventName[0].match(/^\d$/) ||
+      !eventName.match(/^[A-Za-z_0-9]+$/) ||
```
Comment (IjzerenHein) [inline packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts:193]: Perhaps we can provide a cleaner solution as per https://github.com/expo/expo/pull/8391/files ?

File: packages/expo-firebase-analytics/package.json [hunk 2]
```diff
@@ -32,6 +32,9 @@
   "author": "650 Industries, Inc.",
   "license": "MIT",
   "homepage": "https://github.com/expo/expo/tree/master/packages/expo-firebase-analytics",
+  "jest": {
+    "preset": "expo-module-scripts/ios"
```
Comment (IjzerenHein) [inline packages/expo-firebase-analytics/package.json:36]: This appears deprecated. Was there a reason to use the `/ios` preset specifically?

```suggestion
    "preset": "expo-module-scripts"
```


As per `expo-module-scripts/ios/jest-preset`:

```js
const createJestPreset = require('../createJestPreset');

console.warn(
  'The Jest preset "expo-module-scripts/ios" is deprecated, please convert your tests to universal tests and use "expo-module-scripts" instead'
);
module.exports = createJestPreset(require('jest-expo/ios/jest-preset'));
```

**Review by IjzerenHein — CHANGES_REQUESTED** [review]

File: packages/expo-firebase-analytics/package.json (hunk 2 above)
Comment (thorbenprimke) [reply packages/expo-firebase-analytics/package.json:36]: Let me remove `iOS`. Honestly just a copy from another package. 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia2.giphy.com%2Fmedia%2FoCCLHVNt8YO64%2Fgiphy.gif&token=copy" data-gitmeme-token="copy"></a>

File: packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts (hunk 1 above)
Comment (thorbenprimke) [reply packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts:193]: Sure. Perhaps these two sections could also be combined into one method as they have the same logic minus the length.

Comment (frantic) [reply packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts]: Feel free to reuse the regex! I think your PR is more complete, so I've closed mine

File: packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts [hunk 3]
```diff
@@ -172,6 +172,19 @@ class FirebaseAnalyticsJS {
     }
   }
 
+  private static isValidName(name: string, maxLength: number): boolean {
+    return !!(
+      name &&
+      name.length &&
+      name.length <= maxLength &&
+      name.match(/^[A-Za-z][A-Za-z_\d]*$/) &&
+      name !== 'user_id' &&
```
Comment (IjzerenHein) [inline packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts:181]: I think this only applies to the userPropertyName case and not the event-name case (name !== 'user_id')

File: packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts [hunk 4]
```diff
@@ -226,18 +230,9 @@ class FirebaseAnalyticsJS {
...
-      !userPropertyName.length ||
-      userPropertyName.length > 24 ||
-      userPropertyName[0] === '_' ||
-      !userPropertyName.match(/^[A-Za-z_]+$/) ||
-      userPropertyName === 'user_id' ||
-      userPropertyName.startsWith('firebase_') ||
-      userPropertyName.startsWith('google_') ||
-      userPropertyName.startsWith('ga_')
-    ) {
+    if (!FirebaseAnalyticsJS.isValidName(userPropertyName, 24)) {
```
Comment (IjzerenHein) [inline packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts:233]: Perhaps we should add the check for the reserved `user_id` name explicitly here.

```suggestion
    if (!FirebaseAnalyticsJS.isValidName(userPropertyName, 24) || (userPropertyName === 'user_id')) {
```

**Review by IjzerenHein — COMMENTED** [review]
Looks great, many thanks for the unit-tests❤️

File: packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts (hunk 3 above)
Comment (thorbenprimke) [reply packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts:181]: Ah. Let me add this to the unit tests as well. 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2FADr35Z4TvATIc%2Fgiphy.gif&token=faceplam" data-gitmeme-token="faceplam"></a>

**Comment by thorbenprimke** [comment]
@IjzerenHein - updated and added a `user_id` case to each of the tests.

File: packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts (hunk 3 above)
Comment (IjzerenHein) [reply packages/expo-firebase-analytics/src/FirebaseAnalyticsJS.ts]: Haha, sorry for being so nit-picky man 😅

**Review by IjzerenHein — APPROVED** [review]

**Pull request merged**